rubix-core/src/main/java/com/qubole/rubix/core/CachingInputStream.java

```diff
@@ -18,10 +18,9 @@
 import com.google.common.util.concurrent.ListenableFuture;
 import com.google.common.util.concurrent.ListeningExecutorService;
 import com.google.common.util.concurrent.MoreExecutors;
-import com.qubole.rubix.bookkeeper.BookKeeperClient;
 import com.qubole.rubix.bookkeeper.Location;
+import com.qubole.rubix.bookkeeper.RetryingBookkeeperClient;
```

Comment:
Looking at all the issues we have seen regarding dependency orders, we should use interfaces instead.
So instead of using RetryingBookkeeperClient in this class we should be using BookKeeperService.Client.

That will help us in future too if we extract out thrift interfaces to spi.